### PR TITLE
add api zoo_state and zoo_recv_timeout

### LINKED ISSUE #2: ZOOAPI: zoo_recv_timeout
ZOOAPI: int zoo_recv_timeout(zhandle_t *zh);

### PR DESCRIPTION
**Pull request by Yu-zh**
This fixes #2 and #3

### CODE REVIEW

File: src/ZooKeeper.hs [hunk 1]
```diff
@@ -463,6 +465,17 @@ zooWatchGetChildren2 zh path watchfn strsStatfn =
 zooGetClientID :: T.ZHandle -> IO T.ClientID
 zooGetClientID = I.c_zoo_client_id
 
+-- | Get the state of the zookeeper connection
+--
+-- The return valud will be one of the State Consts
+zooState :: T.ZHandle  -> IO CInt
```
Comment (4eUeP) [inline src/ZooKeeper.hs:471]: This function should return `ZooKeeper.Types.ZooState` instead of `CInt`

**Review by 4eUeP — CHANGES_REQUESTED** [review]
Please add some tests for your function

File: src/ZooKeeper.hs [hunk 2]
```diff
@@ -463,6 +465,17 @@ zooWatchGetChildren2 zh path watchfn strsStatfn =
 zooGetClientID :: T.ZHandle -> IO T.ClientID
 zooGetClientID = I.c_zoo_client_id
 
+-- | Get the state of the zookeeper connection
+--
+-- The return valud will be one of the State Consts
+zooState :: T.ZHandle  -> IO T.ZooState
+zooState = (I.ZooState <$>) . I.c_zoo_state
```
Comment (4eUeP) [inline src/ZooKeeper.hs:472]: the `c_zoo_state` ffi can be directly written as  `c_zoo_state :: ZHandle -> IO ZooState`, so you do not need wrap it.

**Review by 4eUeP — CHANGES_REQUESTED** [review]

File: src/ZooKeeper.hs (hunk 2 above)
Comment (Yu-zh) [reply src/ZooKeeper.hs]: Make sense

**Review by 4eUeP — APPROVED** [review]

**Pull request merged**